packages/@sanity/vision/src/components/ParamsEditor.tsx

```diff
@@ -25,9 +25,15 @@ export function ParamsEditor(props: ParamsEditorProps) {
   const {onChange} = props
   const {raw: value, error, parsed, valid} = eventFromValue(props.value)
   const [isValid, setValid] = useState(valid)
+  const [init, setInit] = useState(false)
 
   // Emit onChange on very first render
-  useEffect(() => onChange({parsed, raw: value, valid: isValid, error}), [])
+  useEffect(() => {
+    if (!init) {
```

Comment:
Using `setState` is the canonical way in React today for "run exactly once, on mount" use cases.